### ISSUE TITLE
Tray icon not updating after resuming devise fro hibernation

### ISSUE THREAD

**Issue by webdevs-pro** [soleon/Percentage#55]
![image](https://github.com/user-attachments/assets/54876758-2121-4a75-8371-e2390d2e7af5)
After exiting hibernation mode, the tray icon is not updated. The percentage, font and font size are not updated.

**Comment by nitinp420** [comment]
Yes. I have seen the same issue in the latest version 2.1.5 . Please see the below screenshot for reference.
![Icon not updating](https://github.com/user-attachments/assets/0d542d7f-b4e8-41e6-9185-80d4824b7574)

**Comment by imSamiul** [comment]
![image](https://github.com/user-attachments/assets/07bd1d85-6a84-4bfa-b0a2-1c699457673a)
I also have the same issue. 

OS Name	Microsoft Windows 11 Home Single Language
Version	10.0.22631 Build 22631
System Model	Aspire A515-45
Processor	AMD Ryzen 5 5500U with Radeon Graphics, 2100 Mhz, 6 Core(s), 12 Logical Processor(s)
BIOS Mode	UEFI
Installed Physical Memory (RAM)	16.0 GB
Total Physical Memory	15.4 GB
Available Physical Memory	2.79 GB

**Comment by medic17** [comment]
same issue. resetting to default or restarting the app helps. maybe we can update the code to check if there was a resume from hibernation event

**Comment by soleon** [comment]
Hey guys, thank you for reporting the issue.

When this happens, can you go to the "About" page and see if there's a warning displayed there? If you do, please click the copy button and paste the full error details here for further investigation.

Thanks.

![image](https://github.com/user-attachments/assets/307e14f8-0998-42cd-970a-80ab17b8e0f9)

**Comment by webdevs-pro** [comment]
![image](https://github.com/user-attachments/assets/ae003f31-241e-4813-ba24-b99f5013bc3e)
Today after the tray icon got stuck when I tried to open the app I received this message

**Comment by Kindle-Rabbit** [comment]
There're a lot of similar issues related to percentage show. #64 #63. 

Please fixe this issue ASAP otherwise the application has very limited usage. At least my main scenario is to know the CORRCT percentage on the status bar.

**Comment by dpaakash** [comment]
@soleon I don't see any message in the 'About' page

**Comment by larsen0815** [comment]
Same here. No message in 'About' page.

**Comment by larsen0815** [comment]
> ![image](https://private-user-images.githubusercontent.com/14045233/399567009-ae003f31-241e-4813-ba24-b99f5013bc3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyNTg5NjAsIm5iZiI6MTczNjI1ODY2MCwicGF0aCI6Ii8xNDA0NTIzMy8zOTk1NjcwMDktYWUwMDNmMzEtMjQxZS00ODEzLWJhMjQtYjk5ZjUwMTNiYzNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDE0MDQyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2ZTdhNmNiMjYxMDFkNjc1OTE1MDZkYjQ2YTMxZDlkZjU3ZTI4OTY0ZDdmNjA4Nzc5OTM2ZDA5ZTIzNzUzNjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.76IPxlhcwQiRfoz3qB59_ZDr4lQJPNaJvOdyijyR9QQ) Today after the tray icon got stuck when I tried to open the app I received this message. I have 16 gigs of RAM, 3.2 GB available. It is impossible to stop the application, only through the process manager.

See #69

**Comment by webdevs-pro** [comment]
It looks like this bug has been fixed in version 2.1.8

**Comment by soleon** [comment]
﻿Fixed in v2.1.10.

Root cause: a HICON resource leak in WPF-UI's `NotifyIcon` — every `Icon` assignment internally calls `Bitmap.GetHicon()` but never destroys the previous handle, so GDI handles accumulate until allocations start failing. Once that happens, the tray icon stops updating after wake/sleep, the percentage freezes at the last value, and depending on which call path runs out of resources first you can also see `GDI+` exceptions, `RenderTargetBitmap.FinalizeCreation` `COMException`, or "not enough memory" errors.

Reported upstream as lepoco/wpfui#1313.

The fix is a `SafeNotifyIcon` subclass (commit 7e85027) that overrides `OnPropertyChanged` for `IconProperty` and calls `user32!DestroyIcon` on the prior HICON before WPF-UI allocates the next one. Reflection lookups are cached in static fields so there's no per-update overhead. The XAML in `NotifyIconWindow.xaml` was switched to use `controls:SafeNotifyIcon` so every code path that sets `Icon` is covered.

Please update to v2.1.10 — and reopen if it still happens on that version or later.
